bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java

```diff
@@ -365,17 +410,27 @@ public synchronized void addRule(Rule rule) {
                         logger.error("Cannot create timer for rule '{}': {}", rule.getName(), e.getMessage());
                     }
                 }
+            } else if (t instanceof EventEmittedTrigger) {
+                EventEmittedTrigger eeTrigger = (EventEmittedTrigger) t;
+                Set<Rule> rules = triggerEventTriggeredRules.get(eeTrigger.getChannel());
+                if (rules == null) {
+                    rules = new HashSet<Rule>();
+                    triggerEventTriggeredRules.put(eeTrigger.getChannel(), rules);
+                }
+                rules.add(rule);
             }
         }
     }
 
     /**
      * Removes a given rule from the mapping tables of a certain trigger type
-     * 
+     *
      * @param type the trigger type for which the rule should be removed
      * @param rule the rule to add
      */
     public void removeRule(TriggerTypes type, Rule rule) {
+        // TODO MKA: This doesn't seem right, most collections are maps from String -> x, and remove(Rule) is
```

Comment:
Note: It seems that this was never noticed because this method is only called for STARTUP rules...